### PR TITLE
`General`: Add feature toggle for exports

### PR DESCRIPTION
**Pull request by bassner**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).

#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context

During some periods like a very large exam preparing or running it is not a good idea to start other resource intensive processes like exports. However, this can currently not be prevented by admins.

### Description

Added a feature toggle to globally disable all export operations.

### Steps for Testing

Prerequisites:
- 1 Admin

1. Go to the Feature Toggle page in System Administration
2. Disable the "Exports" toggle
3. Check that you can not perform the following operations (buttons should be disabled, and if you really want you can try to fire a request to the server which it should reject):
    - Archive Exams
    - Archive Courses
    - Export submissions of text, modeling or file upload exercises
    - Export programming exercise repositories (all "Download Repo(s)" and "Download exercise" buttons should be disabled)
4. Re-enable the feature toggle and check that the button are available again (and work)

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

File: src/main/webapp/app/shared/feature-toggle/feature-toggle.directive.ts [hunk 1]
```diff
@@ -30,17 +30,23 @@ export class FeatureToggleDirective implements OnInit, OnDestroy {
...
-            this.featureToggleActiveSubscription = this.featureToggleService
-                .getFeatureToggleActive(this.feature)
-                .pipe(
-                    // Disable the element if the feature is inactive.
-                    tap((active) => {
-                        this.featureActive = this.skipFeatureToggle || active;
-                    }),
-                )
-                .subscribe();
+        if (!this.features) {
```
Comment (Strohgelaender) [inline src/main/webapp/app/shared/feature-toggle/feature-toggle.directive.ts:33]: This method is duplicated in `feature-toggle.directive.ts` and `feature-toggle-link.directive.ts`. Can you find a way to remove this duplication?

File: src/main/webapp/app/shared/feature-toggle/feature-toggle.directive.ts [hunk 2]
```diff
@@ -30,17 +30,23 @@ export class FeatureToggleDirective implements OnInit, OnDestroy {
...
+        const featureArray = Array.isArray(this.features) ? this.features : [this.features];
+        if (!featureArray.length) {
+            return;
+        }
+        this.featureToggleActiveSubscription = this.featureToggleService
+            .getFeatureTogglesActive(featureArray)
+            .pipe(
+                // Disable the element if any of the features is inactive.
+                tap((active) => {
+                    console.log(active);
```
Comment (sleiss) [inline src/main/webapp/app/shared/feature-toggle/feature-toggle.directive.ts:45]:
```suggestion
```

**Review by sleiss — COMMENTED** [review]
Code looks good, just found one `console.log`-statement

File: src/main/webapp/app/shared/feature-toggle/feature-toggle.directive.ts (hunk 1 above)
Comment (bassner) [reply src/main/webapp/app/shared/feature-toggle/feature-toggle.directive.ts:33]: The whole directive is basically a duplicate. Actually, editing the toggle link directive isn't in scope of this PR as I'm not using it anywhere in this PR, so I've just removed the modifications on this file.

We can try to combine both directives in general later, not only that single method, but it's out of scope here

**Review by Dominik-Weinzierl — APPROVED** [review]
Tested during the testing session (stage) - works good :+1:

**Review by maximiliansoelch — APPROVED** [review]
Toggle tested during testing session, works as intended.
Code LGTM 👍

**Review by sleiss — APPROVED** [review]
Tested it in the testing session.
Code also looks good.

**Pull request merged**